src/js/binaryen.js-extended.js

```diff
@@ -0,0 +1,56 @@
+// Extended API
+
+// Makes the internal buffer iterable so we can use Uint8Array.from
+if (typeof Symbol !== 'undefined' && Symbol.iterator) {
```

Comment:
The [iterable protocol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#iterable) relies on `Symbol.iterator` and thus for a browser without `Symbol` (ES6 feature), the `BufferWithRandomAccess` is not extended to be iterable.

`[Symbol.iterator]` is how the iterable protocol is implemented. Making `BufferWithRandomAccess` iterable allows using it with `TypedArray.from(iterable)`. There might be a better way to do this on a higher lever, by using TypedArray in some form (wrapped?) in the Binaryen code, avoiding using the internal `BufferWithRandomAccess`. But for now I think this is a good start.